x-pack/plugins/security_solution/server/lib/timeline/routes/timelines/get_timelines/index.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import Boom from '@hapi/boom';
+
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+
+import type { SecuritySolutionPluginRouter } from '../../../../../types';
+import { TIMELINES_URL } from '../../../../../../common/constants';
+
+import { ConfigType } from '../../../../..';
+import { SetupPlugins } from '../../../../../plugin';
+
+import { buildSiemResponse, transformError } from '../../../../detection_engine/routes/utils';
+
+import { buildFrameworkRequest, escapeHatch, throwErrors } from '../../../utils/common';
+import { getAllTimeline } from '../../../saved_object/timelines';
+import { getTimelinesQuerySchema } from '../../../schemas/timelines';
+
+export const getTimelinesRoute = (
+  router: SecuritySolutionPluginRouter,
+  config: ConfigType,
+  security: SetupPlugins['security']
+) => {
+  router.get(
+    {
+      path: TIMELINES_URL,
+      validate: {
+        query: escapeHatch,
+      },
+      options: {
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      try {
+        const frameworkRequest = await buildFrameworkRequest(context, security, request);
+        const queryParams = pipe(
+          getTimelinesQuerySchema.decode(request.query),
+          fold(throwErrors(Boom.badRequest), identity)
+        );
+        const onlyUserFavorite = queryParams?.only_user_favorite === 'true' ? true : false;
+        const pageSize = queryParams?.page_size ? parseInt(queryParams.page_size, 10) : null;
+        const pageIndex = queryParams?.page_index ? parseInt(queryParams.page_index, 10) : null;
+        const search = queryParams?.search ?? null;
+        const sortField = queryParams?.sort_field ?? null;
+        const sortOrder = queryParams?.sort_order ?? null;
+        const status = queryParams?.status ?? null;
+        const timelineType = queryParams?.timeline_type ?? null;
+        const sort =
+          sortField && sortOrder
+            ? {
+                sortField,
+                sortOrder,
+              }
+            : null;
+        let res = null;
+        let totalCount = null;
+
+        if (pageSize == null && pageIndex == null) {
+          const allActiveTimelines = await getAllTimeline(
```

Comment:
let's create an issue to see if we can use search_after from the find of saved object